### DISCUSSION TITLE
Large monorepo optimizations

### DISCUSSION THREAD

**Discussion by Serg-Mois** [nrwl/nx#14870] [General]
Our 'angular' monorepo consists of 1700+ libs ( "projectType": "library", "projectType": "application") .
We didn't change the nx default layout, created libs using 'nx g @nrwl/angular:lib...' and just migrated from version to version using 'nx migrate latest'. And the latest migration from 13 to 15 affects about 10000 files.
All modules have next config files, that just extended configs in root mostly without any difference: 
tsconfig.json, tsconfig.app.json, tsconfig.spec.json, jest.config.ts, .eslintrc.json, test-setup.ts . 

1. Do we really need all these local configs in libs or they can be removed? Can it somehow break future seamless migration to the next nx versions?
2. If library doesn't have tests at all ( spec.ts ), all autogenerated tests infrastructure ( jest config, tsconfig.spec.ts, test-setup.ts) can be removed right?
3. Does anyone tested the maximum possible size of the nx monorepo from the libs amount perspective? ( e.g. build, lint, work in ide without lags, etc ) - Do you have any 'limits'?

**Comment by zackarydev** [comment]
Yup thinking the same thing! 
We have "small" libs with 6 files, and 6 configuration files. Can we just get rid of the duplicate, empty and meaningless `eslintrc` files? 

I understand why the other ones are there but 🤔
